doc/RELEASE_GUIDE.md

```diff
@@ -9,3 +9,79 @@ You can update Play Store releae metadata using `fastlane android upload_metadat
 But that requires credentials necessary for accessing Play Store API. The simpler way is to edit files contained within [`fastlane/metadata`](metadata) and run the following CI job:
 
 https://ci.status.im/job/status-tools/job/update-playstore-metadata/
+
+# Release process
+
+## Coordination
+
+All the coordination is done in the #release channel on discord.
+
+## Create a branch
+
+First we need to create a release branch cut from develop.
+The name of the branch should be `release/a.b.x` where `x` is actually `x`, not 
+a variable, and `a` and `b` are the `MAJOR` and `MINOR` version of the app.
+
+For example, valid branch names are:
+
+`release/1.12.x` or `release/3.12.x`
+
+
+## Update version
+
+We need to update the `VERSION` file to reflect the release we want to build.
+
+## Building 
+
+Jenkins will automatically build this branch and create release candidates.
+Those will go through the QA process and the QA team will let us know when
+tested successfully.
+
+## Bugfixes before app is released
+
+Because both `status-react` and `status-go` might have newer commits that we don't 
+want to include, bug fixes should be applied separately to develop and the release branch.
+The best workflow is to send a PR to develop, get it merged and the cherry-pick on the
+release branch.
+
+If that's not practical for some reason (i.e develop has diverged significantly from release) 
+other workflows are possible.
+
+In order to change status-go, a similar workflow applies.
+First checkout the tag used in `status-go-version.json`.
+
+Create a branch on status-go named: `release/vx.y.z+hotfix.w` where
```

Comment:
Is it really a "hotfix" if we are "hotfixing" an unreleased version of the app? Sounds like just a normal verison.

Reply:
Ah, but I do recall that there was a case when `status-go` already moved and beyond the release version. I guess in that case it makes sense.

Reply:
Well, it depends, technically it might not be a hotfix, but otherwise we'd have to have yet another denomination for this, we can't use normal versions as normal versions are taken up by the `develop` branch.
So in order not to introduce yet another piece of information, for the sake of simplicity I'd stick with `hotfix` (or `patch` if we want or anything that makes sense, as long as it's the same :) )

Reply:
sorry, missed your last comment before replying, yes, that's the main issue